### DISCUSSION TITLE
Proform zoneHR

### DISCUSSION THREAD

**Discussion by Zvirani786** [cagnulein/qdomyos-zwift#1212] [General]
Hello,
I am extremely new to this.
I have a Perform 400 RI and am trying to Code a HR workout. The plan was to have Resistance change up or down to keep me in a specific zone for a specific amount of time. I tried the Example code with the Rows and Duration ect. The code that was posted here.
I had no luck didn't work for me at all, will not initiate the Time remaining tile was always 0 even though duration was set to 30 seconds. Then I looked at some example workouts the one that came with the app. I mixed both codes to make one that works, and got the duration to show up but that's it. Here it is:

<?xml version="1.0" encoding="UTF-8"?>
<Workout_File>
<Workout> 
<SteadyState Duration="30.0" zoneHR="1" loopTimeHR="10"/>
<SteadyState Duration="60.0" zoneHR="2" loopTimeHR="10"/>
<SteadyState Duration="90.0" zoneHR="3" loopTimeHR="10"/>
</Workout>
</Workout_File>

I also noticed the code was case sensitive, if I type "duration" it will not work, but "Duration" will.

At this point it will show duration and count down but ignore the zoneHR and takes me to max resistance and stays there. I also noticed when I go to select the workout there is a graph on the right side that shows change in power, for some reason it shows a power line at 175 straight across. I think its taking me to max resistant to achieve 175 power. So I am stuck here and need some help.

**Comment by Zvirani786** [comment]
` <?xml version="1.0" encoding="UTF-8"?>
<Workout_File>
<Workout> 
<SteadyState Duration="30.0" zoneHR="1" loopTimeHR="10"/>
<SteadyState Duration="60.0" zoneHR="2" loopTimeHR="10"/>
<SteadyState Duration="90.0" zoneHR="3" loopTimeHR="10"/>
</Workout>
</Workout_File>        `

**Reply by cagnulein** [reply]
@Zvirani786 the format is wrong, you mixed up zwo file with the xml file :)

use this one as a reference https://github.com/cagnulein/qdomyos-zwift/blob/master/train-programs-examples/hrzones_example.xml

**Reply by Zvirani786** [reply]
Thank You Soo much, worked like a charm.

**Reply by cagnulein** [reply]
great!
